### PR TITLE
static_mut_refs: use raw pointers to remove the remaining FIXME

### PR DESCRIPTION
**Pull request by RalfJung**
Using `SyncUnsafeCell` would not make a lot of sense IMO.

### CODE REVIEW

**Comment by compiler-errors** [comment]
r? libs or maybe opsem lol (feel free to assign to someone in particular @RalfJung)

File: library/std/src/sys/thread_local/static_local.rs [hunk 1]
```diff
@@ -25,7 +23,8 @@ pub macro thread_local_inner {
             // FIXME(#84224) this should come after the `target_thread_local`
             // block.
             static mut VAL: $t = INIT_EXPR;
-            unsafe { $crate::option::Option::Some(&VAL) }
+            // SAFETY: we only ever create shared references, so there's no mutable aliasing.
+            unsafe { $crate::option::Option::Some(&*addr_of!(VAL)) }
```
Comment (joboet) [inline library/std/src/sys/thread_local/static_local.rs:27]: This macro is never imported:
```suggestion
            unsafe { $crate::option::Option::Some(&*$crate::ptr::addr_of!(VAL)) }
```

Comment (RalfJung) [reply library/std/src/sys/thread_local/static_local.rs]: Ah, I guess this does not actually get built by PR CI...

**Pull request merged**